File: frontend/src/types/risk.ts
export type RiskTier = 'LOW' | 'MEDIUM' | 'HIGH' | 'CRITICAL';

export interface RiskContributingFactor {
  factor_name: string;
  category: 'weather' | 'temporal_trend' | 'agronomic' | 'biosecurity' | 'history';
  points: number;
  description: string;
}

export interface RiskAssessmentDetail {
  disease_risk_score: number;
  pest_risk_score: number;
  water_stress_risk: number;
  overall_crop_risk: number;
  risk_level: RiskTier;
  rule_version: string;
  assessment_timestamp: string;
  explanation_summary: string;
  contributing_factors: RiskContributingFactor[];
  raw_inputs: Record<string, any>;
}

export interface ZoneRiskResponse {
  zone_id: string;
  zone_code: string;
  farm_id: string;
  risk_assessment: RiskAssessmentDetail;
}

export interface FarmRiskSummaryResponse {
  farm_id: string;
  farm_name: string;
  mean_overall_risk: number;
  highest_risk_level: string;
  critical_zones_count: number;
  evaluated_at: string;
  zones_risk: ZoneRiskResponse[];
}

File: backend/app/schemas/risk.py
from datetime import datetime
from typing import Any, Dict, List, Optional
from pydantic import BaseModel, Field

from app.schemas.common import BaseSchema


class RiskContributingFactor(BaseSchema):
    """Specific factor with point contribution and agronomic explanation."""

    factor_name: str
    category: str = Field(description="weather, temporal_trend, agronomic, biosecurity, history")
    points: int = Field(description="Exact score points added or deducted")
    description: str


class RiskAssessmentDetail(BaseSchema):
    """Explainable multi-vector crop health risk assessment breakdown."""

    disease_risk_score: int = Field(ge=0, le=100, description="0-100 Disease Risk Score")
    pest_risk_score: int = Field(ge=0, le=100, description="0-100 Pest Risk Score")
    water_stress_risk: int = Field(ge=0, le=100, description="0-100 Water Stress Risk")
    overall_crop_risk: int = Field(ge=0, le=100, description="0-100 Composite Crop Risk")
    risk_level: str = Field(description="LOW, MEDIUM, HIGH, CRITICAL")
    rule_version: str
    assessment_timestamp: datetime
    explanation_summary: str
    contributing_factors: List[RiskContributingFactor]
    raw_inputs: Dict[str, Any] = Field(default_factory=dict)


class ZoneRiskResponse(BaseSchema):
    """Zone-level explainable risk assessment response."""

    zone_id: str
    zone_code: str
    farm_id: str
    risk_assessment: RiskAssessmentDetail


class FarmRiskSummaryResponse(BaseSchema):
    """Farm-level aggregate risk intelligence across all subdivided zones."""

    farm_id: str
    farm_name: str
    mean_overall_risk: float
    highest_risk_level: str
    critical_zones_count: int
    evaluated_at: datetime
    zones_risk: List[ZoneRiskResponse]
